engine/src/main/java/org/hibernate/search/query/dsl/impl/ConnectedMultiFieldsPhraseQueryBuilder.java

```diff
@@ -74,7 +72,7 @@ public Query createQuery(FieldContext fieldContext) {
 		final String sentence = phraseContext.getSentence();
```

Comment:
Mmmh, the logic below looks rather convoluted.

Why don't you simply do something like that:
```java
if ( queryContext.getQueryAnalyzer().isRemote() ) {
     return new RemotePhraseQuery(sentence);
}
```
?
with RemotePhraseQuery simply storing the sentence to be reused in ToElasticSearch.

This would simplify the logic in ToElasticSearch too.